tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];
+
+    vsprintf(msg, format, args);
+    gdTestAssertMsg(priority != GD_ERROR,
+                    "unexpected GD_ERROR: %s\n", msg);
+}
+
+
+int main()
+{
+    gdImagePtr im;
+    FILE *fp;
+
+    gdSetErrorMethod(error_handler);
+    fp = gdTestFileOpen2("png", "bug00295.png");
+
+    im = gdImageCreateFromPng(fp);
+
+    fclose(fp);
+    gdImageDestroy(im);
+
+    return gdNumFailures();

```

Comment:
There is no guarantee that `error_handler` will be called, and actually libpng < 1.6 wouldn't throw an error (and the behavior might change in a future version again; might be configurable). Therefore I think the test is okay: we're just checking that no `GD_ERROR` will ever be raised.